### ISSUE TITLE
Vokenization: Improving Language Understanding with Contextualized, Visual-Grounded Supervision

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#131]
## Motivation

- Humans learn by visual pointing, that is, correlating a concept with visual object.
- Conventional image-text datasets are small in size, and the distribution of the caption is different from natural language in corpus like Wikipedia. (Hence they refer to it as *grounded language*)

## TL; DR

- There is no token(word)-level aligned text-image data available.
- Training a token conditioned image generator model is also very challenging.
- Instead train a token-image relevance score function from image caption datasets like MS-COCO and *retrieve* appropriate image from some archive.
- Train BERT with MLM + Voken prediction (NLL) task 

## Previous Work

- BERT: MLM + NSP
- VL-BERT, ViL-BERT: incorporate cross-modal understanding into the language model (but marginal or detrimental effect on language modeling capacity)

## Method

![image](https://user-images.githubusercontent.com/46308407/145136214-619b64f2-9f38-419e-b79f-22c2abedd31a.png)

### Limitations of *Grounded Language*

![image](https://user-images.githubusercontent.com/46308407/145136419-4906d2af-e877-42cb-b643-ab1141c9b3ba.png)

- First they are small in volume
- Also the language distribution significantly differs (grounded languages are mostly descriptions of specific object or motion)

### Token-Image Relevance Score Function

- Text encoder: pretrained (+freezed) BERT with 2-layer MLP on top
- Image encoder: pretrained (+freezed) ResNeXT with 2-layer MLP on top
- Similarity is measured with dot product (both embeddings are normalized)
- With image caption datasets like MS-COCO, learn with triplet loss function. (*distance learning*)
- Regard all tokens in the correct caption as positive pair, and tokens from independent caption sentence as negative pair.

### Training Language Model

![image](https://user-images.githubusercontent.com/46308407/145136575-67900d1e-4946-434c-a11d-499a92d9d9be.png)

- On top of conventional MLM task, add voken prediction task with NLL loss. (The set of vokens is predefined as 50,000 images from VG)

![image](https://user-images.githubusercontent.com/46308407/145136662-2cd28cce-3bf3-44e1-88aa-2ffba8f821ef.png)

- Set equal weights in the multi-task objective.

### Improvement over Vanilla BERT
![image](https://user-images.githubusercontent.com/46308407/145136733-e8e8c01d-ba70-4ea8-aa90-bac7c432a8ce.png)

### Do traditional multi-modal models approaches suffice?

![image](https://user-images.githubusercontent.com/46308407/145136784-aa498192-509c-4b36-b741-16368b26cd56.png)
![image](https://user-images.githubusercontent.com/46308407/145136808-b3bf1921-c065-4249-b04c-a6566680d03c.png)

### Effect of contextualized token-level alignment
![image](https://user-images.githubusercontent.com/46308407/145136854-ef6c0c91-bacc-45a3-a484-5c1bdeb0b1b7.png)

### Sample Retrieval Result

![image](https://user-images.githubusercontent.com/46308407/145136895-6e1fcab7-c92d-43d4-a74e-103225f54cf2.png)

- Retrieved images are diverse
- Contextualization disambiguated words like *down* which can have multiple meanings.

## Comments

- Interesting and promising approach for world understanding.
- Although self-training on large language corpus has been regarded as de facto in language models, world understanding requires more than just books.
- By exploring the visual world, the model can better understand semantic concepts, when the visual signal is provided appropriately.
- Providing appropriate visual signal was crucial, because otherwise it can instead hurt the performance of language models.
- Vokenization is a decent approach, and the authors addressed the issues with conventional image caption datasets in terms of language models. Datasets like MS-COCO are widely used in vision domain, but fall short for pure language tasks.
- Like RAG(Retrieval Augmented Generation), retrieval based methods are a promising *next step* for general AI.
